crypto/stack/stack.c

```diff
@@ -250,7 +250,7 @@ void OPENSSL_sk_pop_free(OPENSSL_STACK *st, OPENSSL_sk_freefunc func)
         return;
     for (i = 0; i < st->num; i++)
         if (st->data[i] != NULL)
-            func(st->data[i]);
+            func((char *)st->data[i]);
```

Comment:
I guess there is a philosophical argument about how much like free() your OPENSSL_sk_freefunc is supposed to be.   That is, if you have a const char *foo = malloc(10);, the original owner of foo is allowed to call free(foo), even though free() is prototyped as "void free(void *)" (not "void free(void const *)") -- the owner of the original allocation is allowed (required!) to remove the const in order to deallocate.  So long as that is what's going on here (and nothing else), it seems acceptable.